lib/ace/transport_app.rb

```diff
@@ -52,6 +52,21 @@ def validate_schema(schema, body)
       end
     end
 
+    # returns a hash of trusted facts that will be used
+    # to request a catalog for the target
+    def self.trusted_facts(certname)
+      # assumes that the certname will be a valid FQDN
+      hostname, domain = certname.split('.', 2)
+      trusted_facts = {
+        "authenticated": "remote",
+        "extensions": {},
+        "certname": certname,
+        "hostname": hostname
+      }
+      trusted_facts[:domain] = domain if domain
```

Comment:
From the Puppet spec:
https://puppet.com/docs/puppet/5.3/lang_facts_and_builtin_vars.html#trusted-facts
> domain — the node’s domain, as derived from its validated certificate name. The value can be empty if the certificate name doesn’t contain a fully qualified domain name.

So do we want to add domain to the initialised trusted_facts hash, then set the value here if it exists?

Reply:
"The value can be empty if the certificate name doesn’t contain a fully qualified domain name." So this would suggest that `domain` must always be present as the method requires that certname is a valid FQDN.

Reply:
But `certname` may not always be a valid FQDN. And `domain: ""` behaves the same as no domain being present. but can happily have it as `domain: ""` if yous prefer

Reply:
https://github.com/puppetlabs/ace/pull/26/files#diff-5bb6dbadea974c974791e62f1392e908R58

Reply:
Yea which just means domain will be empty if it is not a valid FQDN. It's a lazy approach to splitting out a host and domain as agreed in grooming - if domain is not present in the cert, i.e. the cert contains no `.`'s then domain will not need to be sent in the request.